packages/url_launcher/url_launcher_linux/linux/url_launcher_plugin.cc

```diff
@@ -60,6 +75,10 @@ FlMethodResponse* can_launch(FlUrlLauncherPlugin* self, FlValue* args) {
     g_autoptr(GAppInfo) app_info =
         g_app_info_get_default_for_uri_scheme(scheme);
     is_launchable = app_info != nullptr;
+
+    if (!is_launchable) {
+      is_launchable = can_launch_url_with_file_scheme(self, url);
```

Comment:
I would expect that a function called `can_launch_url_with_file_scheme` would expect its argument to be a URL with a file scheme, but that's not being checked anywhere AFAICT.

Reply:
This is actually mimicking how `g_app_info_launch_default_for_uri` detects if URI is launchable. 
Internally glibc checks first if there is registered schema for URI and if not, it checks if URI contains file that can be launched. [Source for g_app_info_launch_default_for_uri](https://github.com/GNOME/glib/blob/main/gio/gappinfo.c#L809-L821)


glibc can support [multiple file schemes](https://github.com/GNOME/glib/blob/main/gio/gvfs.c#L208-L214) additional to `file`-scheme, for example [`resource:`](https://github.com/GNOME/glib/blob/main/gio/gvfs.c#L118-L120)-scheme, and these should be checked as well. 
So it might be that in some use cases URI won't be starting with `file`-scheme, but is launchable anyway.
For this reason I think it is important that canLaunch always return true for URIs that are  launchable with launch method.

Reply:
Function renamed to `can_launch_uri_with_file_resource`